implementations/rust/ockam/ockam_command/src/cmd/inlet.rs

```diff
@@ -1,5 +1,79 @@
-use crate::{args::InletOpts, identity, storage};
-use ockam::{identity::*, route, Context, TcpTransport, TCP};
+use crate::session::initiator::{SessionMaintainer, SessionManager};
+use crate::{args::InletOpts, identity, storage, OckamVault};
+use ockam::{identity::*, route, Context, Result, TcpTransport, TCP};
+use ockam_core::{Address, AsyncTryClone, Route};
+use std::time::Duration;
+
+struct ExistingSession {
+    pub _channel: Address,
+    pub inlet_address: Address,
+}
+
+struct InletSessionManager {
+    args: InletOpts,
+    tcp: TcpTransport,
+    identity: Identity<OckamVault>,
+    policy: TrustMultiIdentifiersPolicy,
+    existing_session: Option<ExistingSession>,
+}
+
+impl InletSessionManager {
+    pub fn new(
+        args: InletOpts,
+        tcp: TcpTransport,
+        identity: Identity<OckamVault>,
+        policy: TrustMultiIdentifiersPolicy,
+    ) -> Self {
+        Self {
+            args,
+            tcp,
+            identity,
+            policy,
+            existing_session: None,
+        }
+    }
+}
+
+#[ockam::worker]
+impl SessionManager for InletSessionManager {
+    async fn start_session(&mut self, _ctx: &Context, timeout: Duration) -> Result<Route> {
+        let channel = self
+            .identity
+            .create_secure_channel_extended(
+                route![
+                    (TCP, &self.args.cloud_addr),
+                    format!("forward_to_{}", self.args.alias),
+                    "secure_channel_listener"
+                ],
+                self.policy.async_try_clone().await?,
+                timeout,
+            )
+            .await?;
+
+        let inlet_address = self
+            .tcp
+            .create_inlet(&self.args.inlet_address, route![channel.clone(), "outlet"])
+            .await?;
+
+        self.existing_session = Some(ExistingSession {
+            _channel: channel.clone(),
+            inlet_address,
+        });
+
+        let responder_route = route![channel, "session_responder"];
+
+        Ok(responder_route)
+    }
+
+    async fn stop_session(&mut self, _ctx: &Context) -> Result<()> {
+        if let Some(existing_session) = self.existing_session.take() {
+            // TODO: Stop SecureChannel
```

Comment:
That said, there are a number of todos still -- the code seems to work but I'm going to wait for you to get back to me on whether or not it's okay to land before they are addressed

Reply:
@thomcc it's the only one todo and it's fine to merge

Reply:
K